src/flick/like/views.py

```diff
@@ -44,3 +52,44 @@ def post(self, request, pk):
             self._create_comment_like_notification(user, comment.owner, comment)
         comment_data = CommentSerializer(comment, context={"request": request}).data
         return success_response(comment_data)
+
+
+class LstLikeView(generics.GenericAPIView):
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def _create_lst_like_notification(self, from_user, to_user, lst):
+        from_profile = Profile.objects.get(user=from_user)
+        notif = Notification()
+        notif.notif_type = "list_like"
+        notif.from_user = from_profile
+        notif.to_user = to_user
+        notif.lst = lst
+        notif.save()
+
+    def post(self, request, pk):
+        if not Lst.objects.filter(pk=pk):
+            return failure_response(f"List of id {pk} does not exist.")
+        lst = Lst.objects.get(pk=pk)
+
+        user = request.user
+        if not Profile.objects.filter(user=user):
+            return failure_response(f"{user} must be logged in.")
+        profile = Profile.objects.get(user=user)
+
+        existing_like = lst.likers.filter(liker=profile)
+        if not existing_like:
+            lst.num_likes += 1
+            like = Like()
+            like.lst = lst
+            like.type = "lst_like"
+            like.liker = profile
+            like.save()
+        else:
+            lst.num_likes -= 1
+            existing_like.delete()
+
+        lst.save(update_fields=["num_likes"])
```

Comment:
hm this seems really sketchy, why do you have to do this?

Reply:
I am doing the list likes similar to notification likes. When the like already exists, the post request cancels the like, which decreases the num_likes field. I can also just calculate the num_likes from the size of likers